go/l1client/rollupcontractlib/handler.go

```diff
@@ -0,0 +1,171 @@
+package rollupcontractlib
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/contracts"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+const methodBytesLen = 4
+
+var (
+	// TODO review estimating gas - these should not be static values
+	defaultGasPrice = big.NewInt(20000000000)
+	defaultGas      = uint64(1024_000_000)
+)
+
+type TxHandler interface {
```

Comment:
@otherview If I'm not mistaken, `PackTx()` will always be called from the Eth/client side, and `UnPackTx()` will always be called from the Obscuro side, right? If so, I think it might make sense to split them into two interfaces, and the unpack side of things can go live under `obscuronode` or `obscurocommon`.

Reply:
I can see your point, but wouldn't that mean initializing a very similar object twice ?

I think these have very connected behaviors and should be together, but how are you seeing these ?

Reply:
IMO each component should get the smallest object possible to do its job, for safety. But I don't feel strongly.